doc/heps/hep2.md

```diff
@@ -0,0 +1,372 @@
+
+This can be easily verified serving this little app with `panel serve app.py`, the warning doesn't get displayed. Note it is possible to display it by serving the app with `python -Wdefault -m panel serve app.py`.
+
+```python
+# app.py
+import warnings
+import panel as pn
+
+warnings.warn('Not displayed :(', DeprecationWarning)
+pn.panel('Hello world!').servable()
+```
+
+#### Pytest
+
+Pytest automatically catches all warnings during test execution and displays them at the end of the session. The issues listed above with `DeprecationWarning`s displayed only in a certain scope don't apply in this case; these warnings are always displayed by Pytest.
+
+
+#### Two types of HoloViz users
+
+We can define two types of HoloViz users:
+
+- Library Developers: depend on HoloViz projects to develop their own library (this includes HoloViz projects themselves as they depend on each other, like Panel with Param)
+- Data Analysts: use HoloViz projects to perform some sort of data analysis, often by writing code in a Notebook or in scripts they run directly or by building and serving a Panel app.
+
+When a HoloViz project deprecates a feature and starts emitting a warning, it is possible it will impact both types of users, however they have different expectations when it comes to deprecation warnings:
+
+- Library Developers don't typically wish the warnings to be propagated to their own users, or at the very least not immediately. Instead, they will usually catch the warnings when running their test suite internally. They want to be given sufficient time to update and release their code, before the feature is removed or before the warning starts to be displayed to their own users.
+- Data Analysts should be warned when a feature they use directly is deprecated, but also should be given an easy way to hide these warnings.
+
+`DeprecationWarning` is the warning type intended for other Python developers; it's certainly the warning type used the most in the Python ecosystem. However, as we've just seen, Data Analysts won't always be set up to see this warning.
+
+#### Consequences
+
+HoloViz users who:
+
+- don't test their code with Pytest (quite likely for Data Analysts)
+- and, don't override the default Python warning filters (even more likely)
+
+won't be notified of `DeprecationWarning`s even if their code directly uses a feature emitting this warning when:
+
+- they serve a Panel app with `panel serve`
+- or, run a script/Notebook that imports a utility module/package that calls a deprecated feature emitting `DeprecationWarning`
+
+### Outcome
+
+Based on the fact that a non-negligeable fraction of HoloViz users may entirely miss `DeprecationWarning`s, we suggest adopting a tiered approach:
+
+1. Start by emitting a `DeprecationWarning`, to inform Library Developers, and some Data Analysts.
+2. After 12 months, upgrade to a `FutureWarning`, to inform all users.
```

Comment:
Saying 12 months here seems to conflict with an overall deprecation time of 6 months mentioned elsewhere.

Reply:
Thanks I missed updating the appendix, fixed in https://github.com/holoviz/holoviz/pull/388/commits/6c7a533c8b0f60e77bec612a9d8cc1b1f8842649.